virtual_rainforest/models/animals/animal_cohorts.py

```diff
@@ -180,4 +226,45 @@ def die_individual(self, number_dead: int, carcass_pool: CarcassPool) -> None:
 
         """
         self.individuals -= number_dead
+
         carcass_pool.energy += number_dead * self.mass * ENERGY_DENSITY["meat"]
+
+    def forage_cohort(
+        self,
+        plant_list: list[PlantCommunity],
+        animal_list: list[AnimalCohort],
+        carcass_pool: CarcassPool,
+        soil_pool: PalatableSoil,
+    ) -> float:
```

Comment:
I would try and turn this into a registry approach (probably not right now, but in the longer term). That would need a generic signature of all possible food sources and then each registered diet method would have body code that targeted the diet specific sources. So, you might have a registry dictionary like:

```python
{
    'herbivore': herbivory(self, plants, animals, carcasses, soil,...),
    'carnivore': carnivory(self, plants, animals, carcasses, soil,...), 
}
```

That then is easier for us or other users to extend with different diets. In addition, you can configure the foraging function when the cohort is created in `init`.

```python
self.forage = DIET_REGISTRY[self.functional_group.diet.value]
```

And then `self.forage` should automatically target the right diet by calling the appropriate function from under that `forage` label.

Reply:
That's basically just replacing `self.forage_cohort` with a direct link to the right function rather than having to maintain a lookup system within `self.forage_cohort`.

Reply:
> It's probably more complicated than that, but it was just a thought.

This is a good point. In the not so distant future get_eaten() will sometimes have to be handled by other modules in some way anyhow. I the registry point but I'll try get_eaten() for now and build towards the registry when things are more fleshed out.